docs/userguide/hyperparameter_optimization.md

```diff
@@ -0,0 +1,244 @@
+[Ray Tune](https://docs.ray.io/en/latest/tune/examples/tune-pytorch-lightning.html).
+
+
+## Hyperparameter optimization with Optuna
+Using [Optuna](https://optuna.org/) is a great option for hyperparameter optimization with Darts. Below, we show a minimal example
+using PyTorch Lightning callback for pruning experiments.
+For the sake of the example, we are training a `TCNModel` on a single series, and optimizing (probably overfitting) 
+hyperparameters for the prediction accuracy on a validation set.
+You can also have a look at [this notebook](https://github.com/unit8co/darts/blob/master/examples/17-hyperparameter-optimization.ipynb)
+for a more complete example.
+
+```python
+import numpy as np
+import optuna
+import torch
+from optuna.integration import PyTorchLightningPruningCallback
+from pytorch_lightning.callbacks import EarlyStopping
+from sklearn.preprocessing import MaxAbsScaler
+
+from darts.datasets import AirPassengersDataset
+from darts.models import TCNModel
+from darts.dataprocessing.transformers import Scaler
+from darts.metrics import smape
+from darts.utils.likelihood_models import GaussianLikelihood
+
+# load data
+series = AirPassengersDataset().load().astype(np.float32)
+
+# split in train / validation (note: in practice we would also need a test set)
+VAL_LEN = 36
+train, val = series[:-VAL_LEN], series[-VAL_LEN:]
+
+# scale
+scaler = Scaler(MaxAbsScaler())
+train = scaler.fit_transform(train)
+val = scaler.transform(val)
+
+# define objective function
+def objective(trial):
+    # select input and output chunk lengths
+    in_len = trial.suggest_int("in_len", 12, 36)
+    out_len = trial.suggest_int("out_len", 1, in_len-1)
+
+    # Other hyperparameters
+    kernel_size = trial.suggest_int("kernel_size", 2, 5)
+    num_filters = trial.suggest_int("num_filters", 1, 5)
+    weight_norm = trial.suggest_categorical("weight_norm", [False, True])
+    dilation_base = trial.suggest_int("dilation_base", 2, 4)
+    dropout = trial.suggest_float("dropout", 0.0, 0.4)
+    lr = trial.suggest_float("lr", 5e-5, 1e-3, log=True)
+    include_year = trial.suggest_categorical("year", [False, True])
+
+    # throughout training we'll monitor the validation loss for both pruning and early stopping
+    pruner = PyTorchLightningPruningCallback(trial, monitor="val_loss")
+    early_stopper = EarlyStopping("val_loss", min_delta=0.001, patience=3, verbose=True)
+    callbacks = [pruner, early_stopper]
+
+    # detect if a GPU is available
+    if torch.cuda.is_available():
+        pl_trainer_kwargs = {
+            "accelerator": "gpu",
+            "gpus": -1,
+            "auto_select_gpus": True,
+            "callbacks": callbacks,
+        }
+        num_workers = 4
+    else:
+        pl_trainer_kwargs = {"callbacks": callbacks}
+        num_workers = 0
+
+    # optionally also add the (scaled) year (cyclically encoded) as a past covariate
```

Comment:
"datetime_attribute" is not cyclic. that would be "cyclic" (also not available for years)
```suggestion
    # optionally also add the (scaled) year value as a past covariate
```